README.md

```diff
@@ -200,6 +200,37 @@ new Vue({
 })
 ```
 
+In TypeScript, all built-in lifecycle hools and special methods are declared in the component instance type to enable auto-complete in editors.
+If you want to make it work with custom hooks, you can manually add it by yourself:
+
+```ts
+import Vue from 'vue'
+import { Route, RawLocation } from 'vue-router'
+
+declare module 'vue/types/vue' {
+  // Augment component instance type
+  interface Vue {
+    beforeRouteEnter?(
+      to: Route,
+      from: Route,
+      next: (to?: RawLocation | false | ((vm: V) => any) | void) => void
+    ): void
+
+    beforeRouteLeave?(
+      to: Route,
+      from: Route,
+      next: (to?: RawLocation | false | ((vm: V) => any) | void) => void
+    ): void
+
+    beforeRouteUpdate?(
+      to: Route,
+      from: Route,
+      next: (to?: RawLocation | false | ((vm: V) => any) | void) => void
+    ): void
+  }
+}
+```
+
```

Comment:
@ktsn In which file should this code be added?

Reply:
Create a new `.ts` file in your project and write in it. You may want to see Vue.js TypeScript guide. https://vuejs.org/v2/guide/typescript.html#Augmenting-Types-for-Use-with-Plugins